weblab/fitting/tests/test_views.py

```diff
@@ -942,3 +978,530 @@ def test_fittingspec_transfer_success(self, client, logged_in_user, other_user,
+        response = client.get('/fitting/specs/%d/results' % public_fittingspec.pk)
+        assert response.status_code == 200
+        assert response.context['fittingspec'] == public_fittingspec
+
+    def test_raises_404_for_non_visible_spec(self, client, private_fittingspec):
+        response = client.get('/fitting/specs/%d/results' % private_fittingspec.pk)
+        assert response.status_code == 404
+
+
+@pytest.mark.django_db
+class TestFittingSpecResultsMatrixJsonView:
+    def test_matrix_json_for_fitting_spec(self, client, public_fittingspec):
+        response = client.get('/fitting/specs/%d/results/matrix' % public_fittingspec.pk)
+        assert response.status_code == 200
+        data = json.loads(response.content.decode())
+        assert 'getMatrix' in data
+
+    def test_raises_404_for_non_visible_spec(self, client, private_fittingspec):
+        response = client.get('/fitting/specs/%d/results/matrix' % private_fittingspec.pk)
+        assert response.status_code == 404
+
+    def test_matrix_json(self, client, quick_fittingresult_version):
+        fit = quick_fittingresult_version.fittingresult
+
+        response = client.get('/fitting/specs/%d/results/matrix?subset=all' % fit.fittingspec.pk)
+        data = json.loads(response.content.decode())
+        assert 'getMatrix' in data
+
+        assert len(data['getMatrix']['rows']) == 1
+        assert str(fit.model_version.sha) in data['getMatrix']['rows']
+        assert len(data['getMatrix']['columns']) == 1
+        assert str(fit.dataset.id) in data['getMatrix']['columns']
+        assert len(data['getMatrix']['experiments']) == 1
+        assert str(fit.pk) in data['getMatrix']['experiments']
+
+        fit_data = data['getMatrix']['experiments'][str(fit.pk)]
+        assert fit_data['id'] == quick_fittingresult_version.id
+        assert fit_data['entity_id'] == fit.id
+        assert fit_data['latestResult'] == quick_fittingresult_version.status
+        assert fit_data['dataset']['id'] == fit.dataset.id
+        assert '/fitting/results/%d/versions/%d' % (fit.id, quick_fittingresult_version.id) in fit_data['url']
+
+    def test_anonymous_cannot_see_private_data(self, client, quick_fittingresult_version):
+        fit = quick_fittingresult_version.fittingresult
+        model_version = fit.model_version
+        model_version.visibility = 'private'
+        model_version.save()
+
+        response = client.get('/fitting/specs/%d/results/matrix?subset=all' % fit.fittingspec.pk)
+        data = json.loads(response.content.decode())
+        assert 'getMatrix' in data
+        assert len(data['getMatrix']['rows']) == 0
+        assert len(data['getMatrix']['columns']) == 1
+        assert len(data['getMatrix']['experiments']) == 0
+
+    def test_ignores_unrelated_datasets(self, client, helpers):
+        related_protocol = recipes.protocol.make()
+        helpers.add_version(related_protocol, visibility='public')
```

Comment:
We can probably get away with `add_fake_version` here (and most/all places below?) to make the tests faster.